draft-yasskin-dispatch-bundled-exchanges.md

```diff
@@ -0,0 +1,675 @@
+* A **wait for N bytes** operation, which blocks until the available bytes
+  include at least N bytes past the current offset. A wait operation fails if
+  the stream ends before enough bytes are received, either due to a network
+  error or because the stream has a finite length.
+* A **seek** operation to change the current offset, relative to either the
+  beginning of the available bytes or to the old current offset. A seek past the
+  end of the available bytes is an *error in this specification*.
+* A **read N bytes** operation, which returns the sequence of N bytes starting
+  at the current offset, and then seeks forward by N bytes. A read past the end
+  of the available bytes is also an error in this specification.
+
+## Load a bundle's metadata {#semantics-load-metadata}
+
+This takes the bundle's stream and returns a struct ({{INFRA}}) of metadata
+containing at least items named:
+
+requests
+
+: A map ({{INFRA}}) whose keys are the HTTP requests ({{FETCH}}) for the
+  exchanges in the bundle, and whose values are opaque metadata that
+  {{semantics-load-response}}{:format="title"} can use to find the matching
+  response.
+
+manifest
+
+: The URL of the bundle's manifest(s). This is a URL to support bundles with
+  multiple different manifests, where the client uses content negotiation to
+  select the most appropriate one.
+
+The struct may include other items added by sections defined in the
+{{section-name-registry}}{:format="title"}.
+
+This operation only waits for a prefix of the stream that, if the bundle is
+encoded with the "responses" section last, ends before the first response.
+
+This operation's implementation is in {{load-metadata}}.
+
+### Load a bundle's metadata from the end {#semantics-load-metadata-from-end}
+
+If a bundle's bytes are embedded in a longer sequence rather than being
+streamed, a parser can also load them starting from a pointer to the last byte
+of the bundle. This returns the same data as {{semantics-load-metadata}}.
+
+This operation's implementation is in {{from-end}}.
+
+## Load a response from a bundle {#semantics-load-response}
+
+This takes the sequence of bytes representing the bundle and one request
+returned from {{semantics-load-metadata}} with its metadata, and returns the
+response ({{FETCH}}) matching that request.
+
+This operation can be completed without inspecting bytes other than those that
+make up the loaded response, although higher-level operations like proving that
+an exchange is correctly signed ({{I-D.yasskin-http-origin-signed-responses}})
+may need to load other responses.
+
+Note that this operation uses the metadata for a particular request returned by
+{{semantics-load-metadata}}, while a client will generally want to load the
+response for a request that the client generated. This specification does not
+define how a client determines the best available bundled response, if any, for
+that client-generated request.
```

Comment:
Why not? We define this for the Cache API, I think ideally we use the same algorithm here.